src/ladds/particle/Constellation.cpp

```diff
@@ -12,68 +12,91 @@
 #include <iostream>
 #include <string>
 
+size_t Constellation::idBase = 1000000;
+
 std::mt19937 Constellation::generator{42};
 
-Constellation::Constellation(const std::string &constellation_data_str,
-                             size_t interval,
-                             double altitudeDeviation,
-                             double coefficientOfDrag)
-    : interval(interval), altitudeDeviation(altitudeDeviation), distribution(0., altitudeDeviation) {
-  // split the 3 comma seperated arguments
-  auto seperator1 = constellation_data_str.find(',', 0);
-  auto seperator2 = constellation_data_str.find(',', seperator1 + 1);
-
-  // argDirPath s1 argStartTime s2  argDuration    // n: 4
-  // a   b   c   d   ,   e   f   ,   g   h   i   // n: 7-4-1=2
-  // 0   1   2   3   4   5   6   7   8   9   10  // n: 11-7-1=3
-  std::string argConstellationName = constellation_data_str.substr(0, seperator1);
-  std::string argStartTime = constellation_data_str.substr(seperator1 + 1, seperator2 - seperator1 - 1);
-  std::string argDuration =
-      constellation_data_str.substr(seperator2 + 1, constellation_data_str.size() - seperator2 - 1);
-
-  // set variables using 3 args
+Constellation::Constellation(ConfigReader &constellationConfig, ConfigReader &config)
+    : constellationName(constellationConfig.get<std::string>("constellation/name")),
+      interval(config.get<size_t>("io/constellationFrequency", 1)),
+      deltaT(config.get<double>("sim/deltaT")),
+      /* altitudeSpread = 3 * sigma */
+      distribution(std::normal_distribution<double>(0, config.get<double>("io/altitudeSpread", 0.0) / 3.0)) {
+  auto coefficientOfDrag = config.get<double>("sim/prop/coefficientOfDrag");
+
+  // set constellations insertion start and duration
+  setStartTime(constellationConfig.get<std::string>("constellation/startTime"),
+               config.get<std::string>("sim/referenceTime", "2022/01/01"));
+  setDuration(constellationConfig.get<std::string>("constellation/duration"));
+
   std::vector<Particle> sats =
-      readDatasetConstellation(std::string(DATADIR) + argConstellationName + "/pos_" + argConstellationName + ".csv",
-                               std::string(DATADIR) + argConstellationName + "/v_" + argConstellationName + ".csv",
+      readDatasetConstellation(std::string(DATADIR) + constellationName + "/pos_" + constellationName + ".csv",
+                               std::string(DATADIR) + constellationName + "/v_" + constellationName + ".csv",
                                coefficientOfDrag);
-
-  // convert vector to deque
+  // convert vector to deque and add noise to altitude
   constellationSize = sats.size();
```

Comment:
Can you explain the difference between `constellationSize` and `satellites.size()` in the doc of `constellationSize`?